ff-fft/src/domain_seq.rs

```diff
@@ -0,0 +1,422 @@
+//! This module contains an `EvaluationDomain` abstraction for
```

Comment:
Do we need this file? I think we can just `#cfg[feature = "std"]` out anything that depends on `rayon`.

Reply:
you are right, after simplifying the code in #86 it is much easier to just add `#cfg[feature = "std"]` everywhere it is needed

Reply:
To clarify, I meant that we can use `#cfg[feature = "std"]` on the original `domain.rs`.